packages/runtime/container-runtime/src/gc/gcHelpers.ts

```diff
@@ -202,3 +205,27 @@ export async function getSnapshotDataFromOldSnapshotFormat(
 		? undefined
 		: { gcState, tombstones: undefined, deletedNodes: undefined };
 }
+
+export function getBaseGCDetailsFromSnapshotData(
+	snapshotData: IGarbageCollectionSnapshotData | undefined,
+): IGarbageCollectionDetailsBase {
+	if (snapshotData === undefined) {
+		return {};
+	}
+
+	let gcData: IGarbageCollectionData | undefined;
+	if (snapshotData.gcState !== undefined) {
+		gcData = { gcNodes: {} };
+		for (const [nodeId, nodeData] of Object.entries(snapshotData.gcState.gcNodes)) {
+			gcData.gcNodes[nodeId] = Array.from(nodeData.outboundRoutes);
+		}
+	}
+	// Run GC on the nodes in the base summary to get the routes used in each node in the container.
+	// This is an optimization for space (vs performance) wherein we don't need to store the used routes of
+	// each node in the summary.
+	const usedRoutes = gcData
```

Comment:
Readability:  I would put everything under the one if block `if (snapshotData.gcState !== undefined)` and in the `else` case do `return { };`  Instead of cascading this ternary operation based on `gcData` which is tied to the result of that `if` check.

(I think I got that right).